src/manager_impl/TaxiPoolManagerImpl.java

```diff
@@ -0,0 +1,38 @@
+package manager_impl;
+
+import entity.Car;
+import manager.TaxiPoolManager;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class TaxiPoolManagerImpl implements TaxiPoolManager {
```

Comment:
`calculateTotalCost `and `findTaxiBySpeed` methods can be rewritten with Java 8 stream API.